### PR TITLE
Mark questions completed

### PR DESCRIPTION
**Pull request by DavidBiddle**
#### What problem does the pull request solve?
Adds the ability for a user to mark a form's questions as complete. 

Trello card: https://trello.com/c/pd1oQb8M/186-forms-admin-allow-form-admin-users-to-mark-the-question-task-as-completed

Found organising this a bit tricky, but my approach here was:
- move the existing 'index' controller action into a new file to stop the pages controller getting too bloated, and add the new action for handling the form into the same controller.
- add a relatively thin MarkCompleteComponent to cover the UI. With some small changes this should be relatively easy to reuse if we want to add this 'mark completion' functionality to other tasks in the list later on.
- added a new feature flag - `task_list_statuses`. This feature doesn't really make sense without the rest of the task list work, so this feature flag should cover all of the task list status work.

#### Checklist

- [x] I've used the pull request template
- [x] I've linked this PR to the relevant issue (if mission work)
- [x] I've written unit tests for these changes (if code change)
- [n/a] I've updated the documentation in (If any documentation requires updating)
    - [n/a] README.md
    - [n/a] Elsewhere (please link)

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 # Used to add feature flags in the app to control access to certain features.
 features:
   make_question_optional: true
```
Comment (aliuk2012) [inline config/settings.yml:3]: We should have a ticket to remove this feature flag  and any conditions

File: spec/components/mark_complete_component/view_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+RSpec.describe MarkCompleteComponent::View, type: :component do
+  let(:mark_complete_form) { Forms::MarkCompleteForm.new(form:).assign_form_values }
+
+  context "when the form has pages" do
+    let(:form) { build(:form, :with_pages, id: 2) }
+
+    context "when the task status feature is enabled" do
+      before do
+        allow(FeatureService).to receive(:enabled?).with(:task_list_statuses).and_return(true)
+      end
```
Comment (aliuk2012) [inline spec/components/mark_complete_component/view_spec.rb:12]:
```suggestion
    context "when the task status feature is enabled", feature_task_list_statuses: true do
```

File: spec/components/mark_complete_component/view_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        render_inline(described_class.new(form.pages, mark_complete_form, "/"))
+
+        expect(page.text).to have_text("Have you finished editing your questions?")
+      end
+    end
+
+    context "when the task status feature is disabled" do
+      before do
+        allow(FeatureService).to receive(:enabled?).with(:task_list_statuses).and_return(false)
+      end
```
Comment (aliuk2012) [inline spec/components/mark_complete_component/view_spec.rb:24]:
```suggestion
    context "when the task status feature is disabled", feature_task_list_statuses: false do
```

File: spec/requests/page_list_spec.rb [hunk 4]
```diff
@@ -0,0 +1,108 @@
+require "rails_helper"
+
+RSpec.describe "Pages", type: :request do
+  before do
+    User.create!(email: "user@example.com")
+  end
```
Comment (aliuk2012) [inline spec/requests/page_list_spec.rb:6]: Why is this needed?

File: app/controllers/page_list_controller.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+private
+
+  def mark_complete_options
+    [OpenStruct.new(value: "true"), OpenStruct.new(value: "false")]
+  end
+
+  def mark_complete_form_params
+    params.require(:forms_mark_complete_form).permit(:mark_complete)
+  end
+end
```
Comment (aliuk2012) [inline app/controllers/page_list_controller.rb:41]: Could we maybe make this controller a bit more generic? I like the fact that the html for the component has been made generic but this controller feels very specific to pages. Maybe a "MarkSectionCompleted" controller?

**Review by aliuk2012 — COMMENTED** [review]
Just a couple comments/suggestions but otherwise looks really good. let me know if you want to jump on a call to work through these together.

File: spec/requests/page_list_spec.rb (hunk 4 above)
Comment (DavidBiddle) [reply spec/requests/page_list_spec.rb:6]: Not sure actually - this was just moved over from the existing pages controller tests. I'll see if removing it breaks anything

File: config/settings.yml (hunk 1 above)
Comment (DavidBiddle) [reply config/settings.yml:3]: Have made one for that here: https://trello.com/c/RsRgO2kq/187-remove-optional-questions-feature-flag

File: app/controllers/page_list_controller.rb (hunk 5 above)
Comment (DavidBiddle) [reply app/controllers/page_list_controller.rb:41]: @aliuk2012 
I'm in two minds about this - while it would be nice to make it more generic the notion of marking the pages task as complete feels like it should be quite closely coupled to the pages list. I guess the way I was imagining it is that the MarkCompleteComponent might appear on multiple pages, but it would post back to different controllers depending on what route was passed into it.

Does that make sense? This might be somewhere where my lack of knowledge re: Rails patterns is letting me down, but I couldn't work out a way of separating out this logic cleanly.

Comment (aliuk2012) [reply app/controllers/page_list_controller.rb:41]: ok,  lets leave it the way you did it. If we find we need it for MarkSectionComplete for other tasks then we can revisit it. Rule of 3 and keeping things DRY :wink:

**Review by aliuk2012 — APPROVED** [review]
